common/stringop.c

```diff
@@ -10,22 +10,32 @@
 
 const char whitespace[] = " \f\n\r\t\v";
 
-char *strip_whitespace(char *_str) {
-	if (*_str == '\0')
-		return _str;
-	char *strold = _str;
-	while (*_str == ' ' || *_str == '\t') {
-		_str++;
+/**************************************************************************//**
+ *
+ * \brief		Strips the whitespace in front and rear of the string. If whitespace
+ *					is found in the middle of the string, all but one space is removed
+ *					between words.
+ *
+ * \param		str	Pointer to the char string that should be stripped. The string
+ *							is modified. If NULL is passed, no operation is performed.
+ *
+ ******************************************************************************/
+void strip_whitespace(char * restrict const str)
+{
```

Comment:
Might be nice to run clang-tidy/clang-format or some other formatter/format verification tool over commits automatically?